Core/Core/Profile/Settings/PairWithObserverViewController.swift

```diff
@@ -53,17 +68,86 @@ class PairWithObserverViewController: UIViewController, ErrorViewController {
+                } else {
+                    if ExperimentalFeature.studentQRCodePairing.isEnabled {
+                        self?.generateQRCode(pairingCode: response?.code)
+                    } else {
+                        self?.spinner.isHidden = true
+                        self?.displayPairingCode(response?.code)
+                    }
+                }
+            }
+        }
+    }
+
+    func generateQRCode(pairingCode: String?) {
+        let termsAndConditionsToGetAccountIDRequest = GetAccountTermsOfServiceRequest()
+        env.api.makeRequest(termsAndConditionsToGetAccountIDRequest) { [weak self] (response, _, error) in
             performUIUpdate {
                 self?.spinner.isHidden = true
                 if let error = error {
                     self?.showError(error)
                 } else {
-                    self?.displayPairingCode(response?.code)
+                    self?.displayQR(pairingCode: pairingCode, accountID: response?.account_id.value, baseURL: self?.env.api.baseURL)
                 }
             }
         }
     }
 
+    func displayQR(pairingCode: String?, accountID: String?, baseURL: URL?) {
+        guard
+            let code = pairingCode,
+            let accountID = accountID,
+            let host = baseURL?.host
+        else { return }
+
+        var comps = URLComponents(string: "canvas-parent://create-account/create-account/\(accountID)/\(code)")
+        comps?.queryItems = [
+            URLQueryItem(name: "baseURL", value: host),
+        ]
+
+        let input = comps?.url?.absoluteString ?? ""
+        let data = input.data(using: String.Encoding.ascii)
+        guard let qrFilter = CIFilter(name: "CIQRCodeGenerator") else { return }
+        qrFilter.setValue(data, forKey: "inputMessage")
+        let transform = CGAffineTransform(scaleX: 10, y: 10)
+        guard let qrImage = qrFilter.outputImage?.transformed(by: transform) else { return }
+
+        qrCodeImageView.image = UIImage(ciImage: qrImage)
+        qrCodeContainer.isHidden = false
+        codeContainer.isHidden = true
+        codeLabel.text = comps?.url?.absoluteString
+        deepLinkURL = comps?.url
+        let attrStr = NSAttributedString(
+            string: NSLocalizedString("Pairing Code: ", comment: ""),
+            attributes: [
+                NSAttributedString.Key.font: UIFont.scaledNamedFont(.regular20),
+                NSAttributedString.Key.foregroundColor: UIColor.named(.textDarkest),
+            ]
+        )
+
+        let attrStr2 = NSAttributedString(
+            string: pairingCode ?? "",
+            attributes: [
+                NSAttributedString.Key.font: UIFont.scaledNamedFont(.semibold20),
+                NSAttributedString.Key.foregroundColor: UIColor.named(.textDarkest),
+            ]
+        )
+        let mutableAttributedString = NSMutableAttributedString()
+        mutableAttributedString.append(attrStr)
+        mutableAttributedString.append(attrStr2)
```

Comment:
appending strings isn't great for i18n, but I don't know how else you'd do it for attributed strings, so this is fine for now.